server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/chatDb/chatDb.ts

```diff
@@ -639,79 +642,123 @@ export class ChatDatabase {
             messages.pop()
             this.#features.logging.debug('Dropped trailing user message')
         }
+
+        //  Make sure there are alternating user and assistant messages
+        let i = 0
+        while (i < messages.length - 1) {
+            const currentMessageType = messages[i].type
+            const nextMessageType = messages[i + 1].type
+
+            if (currentMessageType === nextMessageType) {
+                this.#features.logging.warn(`Deleting message at index ${i + 1} to maintain alternating pattern`)
+                messages.splice(i + 1, 1)
+                continue
+            }
+            i++
+        }
     }
 
-    private validateToolUses(messages: Message[], newUserMessage: ChatMessage): boolean {
+    validateHistoryToolResults(messages: Message[]): void {
         if (messages.length === 0) {
-            if (newUserMessage.userInputMessage?.userInputMessageContext?.toolResults) {
-                this.#features.logging.debug('No history message found, but new user message has tool results.')
-                newUserMessage.userInputMessage.userInputMessageContext.toolResults = undefined
-                // tool results are empty, so content must not be empty
-                newUserMessage.userInputMessage.content = 'Conversation history was too large, so it was cleared.'
+            return
+        }
+
+        let i = 0
+        while (i < messages.length) {
```

Comment:
do we need to look from the beginning?

Reply:
No we don't. No longer trying to account for existing chat tabs with invalid histories. Will only check last Message in history and new UserMessage